### ISSUE TITLE
SGMV not working for llama 70b

### ISSUE THREAD

**Issue by markovalexander** [predibase/lorax#308]
### System Info

```
lorax-launcher --env
2024-03-06T13:20:49.195306Z  INFO lorax_launcher: Runtime environment:
Target: x86_64-unknown-linux-gnu
Cargo version: 1.74.0
```

nvdiia-smi: 

```
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.104.05             Driver Version: 535.104.05   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA A100-SXM4-80GB          On  | 00000000:0F:00.0 Off |                    0 |
| N/A   53C    P0             271W / 400W |  79931MiB / 81920MiB |    100%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
|   1  NVIDIA A100-SXM4-80GB          On  | 00000000:15:00.0 Off |                    0 |
| N/A   53C    P0             377W / 400W |  80075MiB / 81920MiB |    100%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
|   2  NVIDIA A100-SXM4-80GB          On  | 00000000:50:00.0 Off |                    0 |
| N/A   52C    P0             382W / 400W |  77865MiB / 81920MiB |    100%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
|   3  NVIDIA A100-SXM4-80GB          On  | 00000000:53:00.0 Off |                    0 |
| N/A   56C    P0             434W / 400W |  79931MiB / 81920MiB |    100%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
```

model being used: llama-2-70b 

### Information

- [X] Docker
- [ ] The CLI directly

### Tasks

- [X] An officially supported command
- [ ] My own modifications

### Reproduction

Just build the docker from main and try to infer lora adapter for 70b llama model with rank 8 for q.k.v.o matrices. 

I prepared a script to create folder with similar _adapter_ (just tensors of same shape). I attached [script](https://github.com/predibase/lorax/issues/241#issuecomment-1967433603) and details to previous issue. 

There is also an issue with very slow adapter loading: it takes more than 20seconds to load 30mb adapter, which is too slow (adapter is read from local ssd, no other tasks are running on the same node). 

It leads to significant increase in latency, meaning that (even with fast adapter loading) your latency for 1st token with 0.1 qps is more than 15seconds for 5 adapters, which is incredibly slow. 

### Expected behavior

SGMV kernels are utilised and lora inference works fast. 
Adapter weights are sharded in less than 200ms and not 20seconds

**Comment by tgaddair** [comment]
Hi @markovalexander, thanks for providing the details repro script and bug report. I apologize this has taken me longer than I'd hoped to investigate, but I will be carving out some time early this coming week to investigate this throughly.

**Comment by tgaddair** [comment]
Hey @markovalexander, I was able to repro the issue and believe I have a fix (#324). This resolved the issue I was seeing and produced equivalent outputs to the model + adapter running on a single GPU. Apologies it took a while to get this resolved. I'll also be adding some more testing around this code to hopefully catch issues like this before they find their way into releases.
